core/src/main/java/org/radargun/Slave.java

```diff
@@ -46,6 +49,8 @@ private void run(int slaveIndex) throws Exception {
          if (object == null) {
             log.info("Master shutdown!");
             break;
+         } else if (object instanceof ArrayList) {
```

Comment:
I thought Cluster is there only to hold configuration and not real data about the slaves. That's why I didn't add it there. Something more specific than ArrayList would be good probably.